4-Intermediário/8-Interfaces.md

```diff
@@ -0,0 +1,251 @@
+# Interfaces
+Neste tópico abordaremos o assunto do que são interfaces no PHP e como implementá-las
+
+## O que são interfaces? 
+
+<br>
+
+Bom, para entendermos interfaces, vamos pegar uma definição da própria página do PHP:
+
+> Interfaces de objetos permitem a criação de códigos que especificam quais métodos uma classe deve implementar, sem definir como esses métodos serão tratados.
+
+<br>
+
+***Mas então, o que isso significa?*** 
+  
+  - Significa que interfaces são nada mais, nada menos do que modelos de métodos que podemos implementar em uma classe 
+  (ou até mesmo estende-lá em outra interface como veremos mais a seguir), e que ao fazer essa implementação a classe passa a ser obrigada à conter os métodos declarados 
+  na interface, resultando em um erro caso não seja feito;
+  - Em outras palavras, podemos entender interfaces como um contrato, a partir do momento que você assina esse contrato (implementa ele), você passa a ser obrigado a 
+  seguir as especificações previstas nele, o descumprimento delas seria punitivo (que no nosso caso seria um erro fatal do código).
+
+  
+ A partir desse entendimento podemos seguir para o próximo assunto :arrow_down: <br>
+ 
+ ## Como implementar uma interface?
+ 
+ <br>
+ 
+  Ao trabalharmos com interfaces, usaremos principalmente duas palavras reservadas próprias para o seu uso:  ```interface``` e ```implements``` , onde:
+  
+  - **interface:** Palavra reservada para definir a criação de uma interface;
+  - **implements:** Operador usado para realizar a implementação de interfaces;
+
+  Veremos sobre sua síntaxe no código abaixo: 
+  
+```php
+  // sintaxe para a criação:
+  interface nome_interface 
```

Comment:
Acho que pra um melhor entendimento seria legal já de cara colocar exemplos práticos pra essa parte de interfaces. Podendo ser: 
- Comandos
- Provedores
- Factories etc etc

... e também quando for escrever código de exemplo, tente seguir a (PSR-12)[https://www.php-fig.org/psr/psr-12/] e colocar nome classes em **PascalCase**. 

```suggestion
  interface ProvedorPagamentoInterface
```